### ISSUE TITLE
Causing a memory leak (and crash) on Google App Engine

### ISSUE THREAD

**Issue by sudhirj** [disintegration/imaging#5]
I'm trying to use `imaging` in an App Engine app, and after resizing about 6 images the application shuts down because of excessive memory usage. This usually happens because of goroutines that don't exit, so that might be the case. Is there a way to turn off parallel computation via a flag or option for environments like this? Or we can just make sure that all spawned goroutines always exit cleanly.

**Comment by zak905** [comment]
Just for future users who may find this useful, I had also the same issue with `imaging.Blur`, and I got better results by running `runtime.GC()`, The latency is still volatile though and depends on whether the server handles other requests at the same time. After doing some profiling, I saw that `dst := image.NewNRGBA(image.Rect(0, 0, src.w, src.h))` at https://github.com/disintegration/imaging/blob/master/effects.go#L36 eats about 80% of the memory that was allocated (111MB of 138MB). Since there is not much to do besides `runtime.GC()`, I am thinking of trying `gocv` which is a go wrapper for opencv. It wrapps the C implementation which hopefully frees memory as soon as the process is done.
